akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -317,6 +317,8 @@ object ClusterSingletonManager {
         case state: CurrentClusterState ⇒ handleInitial(state)
         case MemberUp(m)                ⇒ add(m)
         case MemberRemoved(m, _)        ⇒ remove(m)
+        case MemberDowned(m) if m.uniqueAddress != cluster.selfUniqueAddress ⇒
+          remove(m)
```

Comment:
That is a too scary change.
Note that `MemberDowned` will happen immediately since Down is an external/user action, while in contrast MemberExited is after convergence on leader and such.
Let's ignore MemberDowned in singleton.